src/main/java/uk/nhs/digital/nhsconnect/nhais/service/EdifactToMeshMessageService.java

```diff
@@ -22,6 +37,20 @@ public MeshMessage toMeshMessage(TranslatedInterchange translatedInterchange) {
         }
         meshMessage.setWorkflowId(WorkflowId.REGISTRATION);
         meshMessage.setContent(translatedInterchange.getEdifact());
+        meshMessage.setMessageSentTimestamp(timestampService.formatInISO(timestampService.getCurrentTimestamp()));
+        return meshMessage;
+    }
+
+    public MeshMessage fromEdifactString(String edifactString, String messageId) {
+        Interchange interchange = edifactParser.parse(edifactString);
+        //send to inbound queue
+        MeshMessage meshMessage = new MeshMessage();
+        meshMessage.setContent(edifactString);
+        meshMessage.setOdsCode(meshCypherDecoder.getSender(edifactString));
```

Comment:
We have a TranslatedInterchange class that contains "other metadata needed for processing". These can be set when the interchange is encoded instead of needing parse the things we just contructed.